src/main/java/org/hobbit/core/containerservice/DirectContainerCreator.java

```diff
@@ -0,0 +1,195 @@
+import java.util.UUID;
+import java.util.concurrent.Future;
+import org.apache.commons.io.Charsets;
+import org.hobbit.core.Commands;
+import org.hobbit.core.Constants;
+import org.hobbit.core.components.AbstractBenchmarkController;
+import org.hobbit.core.components.AbstractCommandReceivingComponent;
+import org.hobbit.core.data.StartCommandData;
+import org.hobbit.core.rabbit.RabbitMQUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import com.google.common.util.concurrent.SettableFuture;
+import com.rabbitmq.client.AMQP.BasicProperties;
+
+/**
+ * This class provides the implementation to create container functionality
+ * @author altaf, sourabh, yamini, melisa
+ *
+ */
+public class DirectContainerCreator implements ContainerCreation {
+	
+	private static final Logger LOGGER = LoggerFactory.getLogger(DirectContainerCreator.class);
+	
+	AbstractCommandReceivingComponent directComponent = null;
+	
+	private AbstractBenchmarkController abstractBenchmarkController;
+	
+	public DirectContainerCreator(AbstractBenchmarkController abstractBenchmarkController) {
+		this.abstractBenchmarkController = abstractBenchmarkController;
+	}
+
+	/**
+     * This method creates and starts an instance of the given image using the given
+     * environment variables.
+     * <p>
+     * Note that the containerType parameter should have one of the following
+     * values.
+     * <ul>
+     * <li>{@link Constants#CONTAINER_TYPE_DATABASE} if this container is part
+     * of a benchmark but should be located on a storage node.</li>
+     * </ul>
+     *
+     * @param imageName
+     *            the name of the image of the docker container
+     * @param containerType
+     *            the type of the container
+     * @param envVariables
+     *            environment variables that should be added to the created
+     *            container
+     * @param netAliases
+     *            network aliases that should be added to the created container
+     * @return the Future object with the name of the container instance or null if an error occurred
+     */
+	protected Future<String> createContainer(String imageName, String containerType, String[] envVariables, String[] netAliases) {
+        try {
+            envVariables = abstractBenchmarkController.extendContainerEnvVariables(envVariables);
+
+            abstractBenchmarkController.initResponseQueue();
+            String correlationId = UUID.randomUUID().toString();
+            SettableFuture<String> containerFuture = SettableFuture.create();
+
+            synchronized (abstractBenchmarkController.getResponseFutures()) {
+            	abstractBenchmarkController.getResponseFutures().put(correlationId, containerFuture);
+            }
+            byte data[] = RabbitMQUtils.writeString(
+            		abstractBenchmarkController.getGson().toJson(new StartCommandData(imageName, containerType, abstractBenchmarkController.getContainerName(), envVariables, netAliases)));
```

Comment:
How does this fit to your own implementation of a direct communication? :thinking: